kiwi/storage/subformat/vagrant_base.py

```diff
@@ -0,0 +1,266 @@
+      provide it with the correct base indentation.
+    """
+    def post_init(self, custom_args):
+        """
+        vagrant disk format post initialization method
+
+        store vagrantconfig information provided via custom_args
+
+        :param dict custom_args:
+            Contains instance of xml_parse::vagrantconfig
+
+            .. code:: python
+
+                {'vagrantconfig': object}
+        """
+        if not custom_args or 'vagrantconfig' not in custom_args:
+            raise KiwiFormatSetupError(
+                'object init requires custom_args hash with a vagrantconfig'
+            )
+
+        if not custom_args['vagrantconfig']:
+            raise KiwiFormatSetupError(
+                'no vagrantconfig provided'
+            )
+
+        self.image_format = 'vagrant.' + self.provider + '.box'
+        self.vagrantconfig = custom_args['vagrantconfig']
+
+    def create_image_format(self):
+        # type: () -> None
+        """
+        Create a vagrant box for any provider. This includes:
+
+        * creation of box metadata.json
+        * creation of box Vagrantfile
+        * creation of result format tarball from the files created above
+        """
+        self.temp_image_dir = mkdtemp(prefix='kiwi_vagrant_box.')
+
+        box_img_files = self.create_box_img(self.temp_image_dir)
+
+        metadata_json = os.sep.join([self.temp_image_dir, 'metadata.json'])
+        with open(metadata_json, 'w') as meta:
+            meta.write(self._create_box_metadata())
+
+        vagrantfile = os.sep.join([self.temp_image_dir, 'Vagrantfile'])
+        with open(vagrantfile, 'w') as vagrant:
+            vagrant.write(self._create_box_vagrantconfig())
+
+        Command.run(
+            [
+                'tar', '-C', self.temp_image_dir,
+                '-czf', self.get_target_file_path_for_format(self.image_format),
+                os.path.basename(metadata_json),
+                os.path.basename(vagrantfile)
+            ] + [
+                os.path.basename(box_img_file)
+                for box_img_file in box_img_files
+            ]
+        )
+
+    def store_to_result(self, result):
+        # type: (Result) -> None
+        """
+        Store result file of the vagrant format conversion into the
+        provided result instance. In this case compression is unwanted
+        because the box is already created as a compressed tarball
+
+        :param object result: Instance of Result
+        """
+        result.add(
+            key='disk_format_image',
+            filename=self.get_target_file_path_for_format(
+                self.image_format
+            ),
+            use_for_bundle=True,
+            compress=False,
+            shasum=True
+        )
+
+    @classmethod
+    def get_additional_metadata(cls):
+        # type: () -> Dict[str, str]
+        return {}
+
+    @classmethod
+    def get_additional_vagrant_config_settings(cls):
+        # type: () -> str
+        return ""
```

Comment:
It should just pass and avoid that comment block. When using a template class there is no need for the method to return an empty string. But it should mention that this method can be overwritten by the specialized provider
class. e.g

```python
@classmethod
def get_additional_vagrant_config_settings(cls):
    """
    Handle additional configurations in the Vagrantfile

    Depending on the provider additional vagrant configuration settings are required.
    The implementation of those are done in the specialized provider class.
    """
    pass
```